### DISCUSSION TITLE
RLS policy doesn't work when data accessed via prisma

### DISCUSSION THREAD

**Discussion by gokulkrishh** [supabase/supabase#11674] [Questions] [answered]
I am using NextJS routes which uses Prisma to make API calls, after setting RLS policy, I am still able to access the data.

Policy:

```sql
BEGIN;
  ALTER POLICY "All" ON "public"."subscriptions" USING (uid() = user_id);
  ALTER POLICY "All" ON "public"."subscriptions" WITH CHECK (uid() = user_id);
COMMIT;
``` 

public subscriptions table have user_id associated with public users table. When I call the API via NextJS routes like below

```js
import prisma from 'lib/prisma';

export default async function handle(req, res) {
	try {
		const data = await prisma.subscriptions.findMany();

		return res.status(200).json(data);
	} catch (error) {
		return res.status(500).send({ error: JSON.stringify(error || ''), message: 'Failed to get the all subscriptions' });
	}
}
```

Same data is returned for User1 and User2, Ideally data should returned for User1 and User2 it should be empty.

**Comment by GaryAustin1** [comment, marked as answer]
I don’t believe you can use auth calls with Prisma as it is not coming in thru the REST API and has its own user setup.  I’m not sure it honors RLS.   But I don’t use Prisma. 

https://github.com/supabase/supabase/discussions/4051
and search here or the web for Prisma RLS.

**Comment by gokulkrishh** [comment]
@GaryAustin1  It doesn't honor RLS hence posted the question here, thanks. It looks like #4051 Prisma doesn't use RLS from supabase, so handled it via next js API for now.
